src/model/Building.cpp

```diff
@@ -932,6 +933,30 @@ namespace model {
       return Transformation::rotation(Vector3d(0, 0, 1), -degToRad(this->northAxis()));
     }
 
+    double Building_Impl::exteriorPerimeter() {
```

Comment:
You might see if ASHRAE, BEDES, or some other standard has a definition of exterior perimeter you can reference